core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java

```diff
@@ -767,6 +766,14 @@ private void handleRequestOrResponseComplete() {
                     requestOrResponseComplete = true;
                     return;
                 }
+                if (req.isHttp1WebSocket()) {
```

Comment:
Sorry, I've realized that this won't be reached if the request hits the following lines:
- https://github.com/line/armeria/blob/e08450850800a2022448d3bca57775f7208252d2/core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java#L332
- https://github.com/line/armeria/blob/e08450850800a2022448d3bca57775f7208252d2/core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java#L360

I'm not sure what the best way to handle this is at the moment 😅  Any ideas are welcome

Reply:
Good point. I think we need to call `req.abort()` for the early returned requests. 
Otherwise, `ByteBuf` in the requests could be leaked.

We may decrease the increased metric in `.abort()` when `req.init()` isn't invoked.

Reply:
Thanks for identifying these locations. 👍 
I think we can proceed as follows:
- Increase the pending requests in `HttpRequestDecoder`s regardless of whether requests are aggregating or streaming.
- Decrease the pending requests in the locations identified by @jrhee17.
- Decrease the pending requests and increase the active requests right before invoking `service.serve()`.
What do you think?